### PR TITLE
GL-107: Add Admin API for listing all themes

### PR DESCRIPTION
**Pull request by rasikasri**
### Jira link

[GL-107](https://transformuk.atlassian.net/browse/GL-107)

### What?

I have added/removed/altered:

- [x] Added a GET admin api to list out all themes from the database 

### Why?

I am doing this because:

- This will enable a theme drop down in the category assessments admin area

### Have you? (optional)

- [x] Added documentation for new apis

### CODE REVIEW

File: app/controllers/api/admin/green_lanes/themes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        before_action :check_service, :authenticate_user!
+
+        def index
+          render json: serialize(themes.to_a)
+        end
+
+        private
+
+        def themes
+          @themes ||= ::GreenLanes::Theme.order(Sequel.desc(:id))
```
Comment (jebw) [inline app/controllers/api/admin/green_lanes/themes_controller.rb:14]: This would make more sense as section and subsection

File: app/controllers/api/admin/green_lanes/themes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        end
+
+        def serialize(*args)
+          Api::Admin::GreenLanes::ThemeSerializer.new(*args).serializable_hash
+        end
+
+        def serialize_errors(theme)
+          Api::Admin::ErrorSerializationService.new(theme).call
+        end
+
```
Comment (jebw) [inline app/controllers/api/admin/green_lanes/themes_controller.rb:24]: AFAICT this method is unused - I'm surprised rubocop didn't complain about it?

File: app/serializers/api/admin/green_lanes/theme_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+        set_id :id
+
+        attributes :section,
+                   :subsection,
+                   :theme,
+                   :description,
+                   :category,
+                   :created_at,
+                   :updated_at
```
Comment (jebw) [inline app/serializers/api/admin/green_lanes/theme_serializer.rb:17]: Not convinced we need to be serializing these two? No strong feelings though

File: config/routes.rb [hunk 4]
```diff
@@ -66,6 +66,7 @@
         end
 
         resources :category_assessments, module: 'green_lanes', only: %i[index show create update destroy]
+        resources :themes, module: 'green_lanes', only: %i[index]
```
Comment (jebw) [inline config/routes.rb:69]: Both these routes could be moved under a nested namespace (like news)

File: docker-compose.yml [hunk 5]
```diff
@@ -22,7 +22,7 @@ services:
        - hmrc-postgres13:/var/lib/postgresql/data
   hmrc-opensearch:
     container_name: hmrc-opensearch
-    image: opensearchproject/opensearch:2
+    image: opensearchproject/opensearch:2.11.0
```
Comment (jebw) [inline docker-compose.yml:25]: I'd say either leave this out, or move it to its own PR since its not relevant to this PR

In general I'm inclined to say leave it as is, which forces someone to solve the env var question properly rather than just locking us to an outdated version

File: spec/requests/api/admin/green_lanes/themes_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  describe 'GET to #index' do
+    let(:make_request) do
+      authenticated_get api_admin_themes_path(format: :json)
+    end
+
+    context 'with some themes' do
+      before do
+        theme
+      end
```
Comment (jebw) [inline spec/requests/api/admin/green_lanes/themes_controller_spec.rb:19]: super minor but this is so short it could just be a one liner `before { theme }` ? no strong feelings either way on this

**Review by jebw — CHANGES_REQUESTED** [review]
Couple of minor suggestions and the compose file change can be dropped but looks great

File: app/controllers/api/admin/green_lanes/themes_controller.rb (hunk 1 above)
Comment (rasikasri) [reply app/controllers/api/admin/green_lanes/themes_controller.rb]: done

File: app/controllers/api/admin/green_lanes/themes_controller.rb (hunk 2 above)
Comment (rasikasri) [reply app/controllers/api/admin/green_lanes/themes_controller.rb]: removed

File: config/routes.rb (hunk 4 above)
Comment (rasikasri) [reply config/routes.rb]: done

File: spec/requests/api/admin/green_lanes/themes_controller_spec.rb (hunk 6 above)
Comment (rasikasri) [reply spec/requests/api/admin/green_lanes/themes_controller_spec.rb]: done

**Review by jebw — APPROVED** [review]
LGTM :raised_hands:

**Review by bhavanibandla — APPROVED** [review]
Approved to merge this PR to the main branch.
